lib/scsocketcreator.js

```diff
@@ -47,17 +47,23 @@ function create(options) {
 
   options = options || {};
 
-  if (options.host && options.port) {
-    throw new InvalidArgumentsError('The host option should already include the' +
-      ' port number in the format hostname:port - Because of this, the host and port options' +
-      ' cannot be specified together; use the hostname option instead');
+  if (options.host && !options.host.match(/[^:]+:\d{2,5}/)) { // should be "hostname:port"
+    throw new InvalidArgumentsError('The host option should include both' +
+      ' the hostname and the port number in the format "hostname:port".');
+  }
+
+  if ((options.host && options.port) || (options.host && options.hostname)) {
+    throw new InvalidArgumentsError('The host option should already include' +
+      ' the hostname and the port number in the format "hostname:port"' +
+      ' - Because of this, the host and port options cannot be specified together.' +
+      ' Use either hostname and port options together or host option alone.');
```

Comment:
@MegaGM Maybe this should be two separate if statements with two different error messages?
The part `the host and port options cannot be specified together` is not relevant if the user passed `host` and `hostname`.

Reply:
Also, you'll probably think this is silly nit picking (and you'd be right ;p) but the convention in the rest of the code currently is that error strings are a single sentence without any dot characters - See https://github.com/SocketCluster/socketcluster/blob/master/index.js#L181-L183 - Although I actually like it more with a dot at the end (it's more correct grammatically), I think that it's more important to be consistent.

Reply:
> the host and port options cannot be specified together

Overlooked. My bad.

Reply:
DRY, sir. So 1 `if` is better than 2, I guess, since they are tightly related.